src/service/ReservationService.java

```diff
@@ -0,0 +1,101 @@
+package service;
+
+import model.Customer;
+import model.IRoom;
+import model.Reservation;
+
+import java.util.*;
+
+public class ReservationService {
+    public static final ReservationService reservationService = new ReservationService();
+    private static final List<Reservation> reservations = new ArrayList<>();
+    private static final Map<String, IRoom> rooms = new HashMap<>();
+
+    public Collection<IRoom> findRooms(Date checkInDate, Date checkOutDate) {
+        List<IRoom> availableRooms = new ArrayList<>();
+        for (IRoom room : rooms.values()) {
+            if (reservationService.checkAvailability(room, checkInDate, checkOutDate)) {
+                availableRooms.add(room);
+            }
+        }
+        return availableRooms;
+    }
+
+    public Collection<IRoom> recommendedRooms(Date checkInDate, Date checkOutDate) {
+        System.out.println("No room found in the given date!");
+        Date checkInRecommended = addDaysToDate(checkInDate, 7);
+        Date checkOutRecommended = addDaysToDate(checkOutDate, 7);
+        System.out.println("Recommended room for 7 days later: ");
+        return findRooms(checkInRecommended, checkOutRecommended);
+    }
+
+    public Date addDaysToDate(Date date, int add) {
+        Calendar cal = Calendar.getInstance();
+        cal.setTime(date);
+        cal.add(Calendar.DAY_OF_MONTH, add);
+        return cal.getTime();
+    }
+
+    public IRoom getARoom(String roomNumber) {
+        return rooms.get(roomNumber);
+    }
+
+    public Collection<IRoom> getAllRooms() {
+        return rooms.values();
+    }
+
+    public Reservation reserveARoom(Customer customer, IRoom room, Date checkInDate, Date checkOutDate) {
+        if (reservationService.checkAvailability(room, checkInDate, checkOutDate)) {
+            Reservation newReservation = new Reservation(customer, room, checkInDate, checkOutDate);
```

Comment:
I don't think this is required for your assignment, but be aware that your code is by no means thread safe (for a LOT of reasons), but imagine you call line `48`  which is true. Then, before line 49 or 50 are called it could be false because some other thread reserved the room already.

This is not directly the most important aspect to know about Java, but it will be required to know in any (large) company where you would work with Java.